fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt

```diff
@@ -0,0 +1,109 @@
+
+import android.content.res.Configuration
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.FloatingActionButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.ui.colors.PhotonColors
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Floating action button, to be used inside a [Box], to allow aligning the FAB to the bottom end corner.
+ *
+ * @param modifier [Modifier] to be applied to the action button.
+ * @param icon [Composable] icon to be displayed inside the action button, defaults to new tab icon.
+ * @param label Text to be displayed next to the icon.
+ * @param onClick Invoked when the action button is clicked.
+ */
+@Composable
+fun BoxScope.FloatingActionButton(
+    modifier: Modifier = Modifier,
+    icon: @Composable () -> Unit = {
+        Icon(
+            painter = painterResource(R.drawable.ic_new),
+            contentDescription = if (Theme.getTheme() == Theme.Private) {
+                stringResource(
+                    R.string.add_private_tab,
+                )
+            } else {
+                stringResource(
+                    R.string.add_tab,
+                )
+            },
+            tint = PhotonColors.White,
+        )
+    },
+    label: String? = null,
+    onClick: () -> Unit,
+) {
+    FloatingActionButton(
+        onClick = onClick,
+        modifier = Modifier
+            .align(Alignment.BottomEnd)
+            .then(modifier),
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        contentColor = PhotonColors.White,
+        elevation = FloatingActionButtonDefaults.elevation(99.dp),
+    ) {
+        Row(
+            modifier = Modifier
+                .wrapContentSize()
+                .padding(16.dp)
+                .animateContentSize(),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            icon()
+            if (!label.isNullOrBlank()) {
+                Row {
+                    Spacer(Modifier.width(12.dp))
+                    Text(
```

Comment:
We should set the text color textActionPrimary https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?node-id=1379%3A4460&t=XigaI5op6FGRhkjh-0